spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundGateway.java

```diff
@@ -103,7 +104,7 @@ public void onMessage(Message message, Channel channel) {
 				final org.springframework.messaging.Message<?> reply = sendAndReceiveMessage(request);
 				if (reply != null) {
 					// TODO: fallback to a reply address property of this gateway
-					Address replyTo = message.getMessageProperties().getReplyToAddress();
+					Address replyTo = AddressUtils.decodeReplyToAddress(message);
```

Comment:
Yes; I thought long and hard about this; I thought about closing this and just having the fix in AMQP.

However, in that way, SI won't work with 1.4.1 or 1.4.2.

This way, 4.1.2 will work with 1.4.1, 1.4.2, 1.4.3 (not 1.4.0).

I thought that was the least bad solution.